ui/app/common/displaycontrols/observation/services/formRecordTreeBuildService.js

```diff
@@ -148,16 +155,32 @@ angular.module('bahmni.common.displaycontrol.observation')
                         "conceptClass": null
                     }
                 };
-                obsGroup.concept.shortName = column.value;
+                var translationKey = column.translationKey;
+                var defaultShortName = column.value;
+                obsGroup.concept.shortName = self.getTranslatedShortName(translationData, translationKey, obsGroup, defaultShortName);
                 var columnRecord = self.getColumnObs(index, record);
                 column.children = columnRecord;
-                self.createGroupMembers(column, obsGroup, obsList);
+                self.createGroupMembers(column, obsGroup, obsList, translationData);
                 if (obsGroup.groupMembers.length > 0) {
                     tableGroup.groupMembers.push(obsGroup);
                 }
             });
         };
 
+        self.getTranslatedShortName = function (translationData, translationKey, obsGroup, defaultShortName) {
+            if (self.isTranslationKeyPresent(translationData, translationKey)) {
+                return translationData.labels[translationKey][0];
+            }
+            else {
```

Comment:
HI @vinishad, please remove unnecessary **else** block, you can directly **return** it

Reply:
Sure. thanks agian @sumanmaity112